src/daemons/linuxd/src/linuxd/mod.rs

```diff
@@ -95,19 +100,89 @@ pub struct LinuxDaemon {
 
 impl LinuxDaemon {
     pub fn init(
-        control_plane_stream: SocketStream,
+        control_plane_sockaddr: String,
+        control_plane_socktype: SocketType,
         user_vm_listener: SocketListener,
         gateway_listener: Option<SocketListener>,
+        in_l2: bool,
     ) -> Result<Self, Error> {
         Ok(Self {
             assembler: Arc::new(Mutex::new(RequestAssembler::default())),
-            control_plane_stream,
+            control_plane_sockaddr,
+            control_plane_socktype,
             user_vm_listener,
             gateway_listener,
             venv: Arc::new(Mutex::new(VirtualEnviromentDirectory::new())),
+            in_l2,
         })
     }
 
+    /// This helper method accepts a connection from the control-plane.
+    fn accept_control_plane_connection(&self) -> Result<SocketStream> {
+        match SocketStream::connect(
+            self.control_plane_socktype,
+            self.control_plane_sockaddr.clone(),
+        ) {
+            Ok(socket) => {
+                info!("Connected to control plane on: {:?}", self.control_plane_sockaddr);
+                Ok(socket)
+            },
+            Err(e) => {
+                let reason: String = format!(
+                    "failed to connect to control-plane socket address (address={}, error={e:?})",
+                    self.control_plane_sockaddr.clone()
+                );
+                error!("{reason}");
+                Err(anyhow::anyhow!(reason))
+            },
+        }
+    }
+
+    /// This helper method will trap waiting for a message in a given port if we are about to be
+    /// snapshotted.
+    fn trap_if_pending_snapshot(&self) -> Result<()> {
+        if self.in_l2 {
+            // We only need one token to block in the trap poll.
+            const TRAP_TOKEN: Token = Token(0);
+
+            let mut trap_listener: SocketListener =
+                Socket::bind(SocketType::Tcp, restore_gate_sockaddr_builder())?;
+
+            let mut trap_poll: Poll = Poll::new()?;
+            trap_poll
+                .registry()
+                .register(&mut trap_listener, TRAP_TOKEN, Interest::READABLE)?;
+
+            let mut events: Events = Events::with_capacity(config::syscomm::MAX_NUM_POLL_EVENTS);
+            'poll_loop: loop {
+                // Deliberately print to stdout so that it can be captured by the snapshot creation
+                // script.
+                println!("Linuxd hit pre-snapshot trap");
+
```

Comment:
This string constant should be synchronized with the `generate-l2-snapshot` script. Ideally, we would read it from a common source, for instance a build-time environment variable.